src/win32_window.c

```diff
@@ -65,19 +74,43 @@ static int getAccentColor(float color[4])
     UINT dwImmersiveColorType = _glfw.win32.uxtheme.GetImmersiveColorTypeFromName(L"ImmersiveSystemAccent");
     UINT dwImmersiveColorSet = _glfw.win32.uxtheme.GetImmersiveUserColorSetPreference(FALSE, FALSE);
 
-    UINT rgba = _glfw.win32.uxtheme.GetImmersiveColorFromColorSetEx(dwImmersiveColorSet,
+    UINT abgr = _glfw.win32.uxtheme.GetImmersiveColorFromColorSetEx(dwImmersiveColorSet,
                                                                     dwImmersiveColorType,
                                                                     FALSE,
                                                                     0);
 
-    color[0] = (float) (0xFF & rgba);
-    color[1] = (float) ((0xFF00 & rgba) >> 8);
-    color[2] = (float) ((0xFF0000 & rgba) >> 16);
-    color[3] = (float) ((0xFF000000 & rgba) >> 24);
+    // color is in RGBA
+    color[0] = (float) (0xFF & abgr) / 255.f;
+    color[1] = (float) ((0xFF00 & abgr) >> 8) / 255.f;
+    color[2] = (float) ((0xFF0000 & abgr) >> 16) / 255.f;
+    color[3] = (float) ((0xFF000000 & abgr) >> 24) / 255.f;
     
     return GLFW_TRUE;
 }
 
+// Set a custom accent color for a window
+//
+static int setAccentColor(HWND handle, const float color[4])
+{
+    if (!_glfw.win32.uxtheme.uxThemeAvailable)
+        return GLFW_FALSE;
+
+    // The alpha should be left as zero, otherwise it won't work
+    COLORREF accentColor = (((UINT) (color[2] * 255)) << 16) | (((UINT) (color[1] * 255)) << 8) | ((UINT) (color[0] * 255));
+
+    DwmSetWindowAttribute(
+            handle, DWMWA_BORDER_COLOR,
+            &accentColor, sizeof(accentColor)
+    );
+
+    DwmSetWindowAttribute(
+            handle, DWMWA_CAPTION_COLOR,
+            &accentColor, sizeof(accentColor)
+    );
+
+    return GLFW_TRUE;
+}
+
```

Comment:
How do you reset this color to the defaults?

According to the documentation: 
> The app is responsible for changing the border color according to state changes, such as a change in window activation. 
How can GLFW handle that?

Reply:
@ws909 Good question. I wasn't aware of that. I have added a new function to do that. A dedicated function might not be necessary, as we can achieve the same result by passing a specific color to `setAccentColor`. It's up to you.

Reply:
Thanks! :)

One more question: would it suffice to observe `WM_SETTINGCHANGE` in the helper window’s event listening function to support the system theme changed callback?